### ISSUE TITLE
Getting search terms from an object - Re-opened

### ISSUE THREAD

**Issue by notdanross** [nss-day-cohort-46/How-to-Ask-for-Help#139]
**_Ticket MUST be completed before getting help from an instructor_**
#### Have you searched the other issue tickets? If not, do that first. 
- [x] Yes

####  REQUIRED Add Tags to Issue Ticket and Assign to Project Board
- [x] Added to `c46 Help-Thread` project board
- [x] Added the `Blocker` label if this is stoping you from moving forward

#### REQUIRED Describe your issue:
>> For my capstone I copied an array of objects from an api and added integer ids to use as primary keys.
I am now trying to populate a drop-down with a list of items from each object, but I need to exclude certain keys as they are not features. Unsure of the best way to approach this. 

#### REQUIRED Copy/paste any error messages that appear:
```Unhandled Rejection (TypeError): Cannot convert undefined or null to object```
My code is expecting an object. Tried using .then to run the code after a fetch call but it has not worked. 

#### REQUIRED What have you googled? **_(You must have at least 3 links related to your issue)_**
- [x] 1. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys
- [x] 2. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values
- [ ] 3. 

#### REQUIRED: What have you tried? Tell us everything you have tried. 
>>
I'm waiting to get started on this specific piece of logic. I've considered:
- [ ] using Object.keys to grab all the keys, then add hard coded logic to filter out unwanted keys

#### REQUIRED: Copy/paste link TO THE FILE on your branch that you are having issues with
[My database](https://github.com/Daniel-L-Ross/parkbook-api/blob/main/database.json)
[ParkSearch](https://github.com/Daniel-L-Ross/parkbook/blob/dr-filterParks/src/components/parks/ParkSearch.js) (where I will iterate over something to populate the dropdown) 

#### Code Snippet

```
    useEffect(() => {
        getParks()
            .then(() =>{
                let featureArray = []
                debugger
                Object.values(parks[0]).map(feature => {
                    if (feature === "Yes" || "No") {
                        const prettyFeature = parks[0][feature].replace(/_/g, ' ')
                        featureArray.push(prettyFeature)
                    }
                }) 
                setFeatures(featureArray)
            })
    }, [])
```

**Comment by jakefroeb** [comment]
Interesting problem... if you plan to use the api later as a stretch goal then i would code some logic to go through the keys and only display the ones you want, otherwise i would consider modifying your data to hold only the key value pairs you would use.

**Comment by Scott47** [comment]
@jakefroeb @Daniel-L-Ross 's solution is very specific to his data, but you should check out his solution.

**Comment by notdanross** [comment]
@jakefroeb this api is actually the basis of my capstone, so I will need a working solution relatively early in the development process. I had the same thought of hard-coding what values to exclude, but I wanted a solution that would work for a very large object.
Modifying the data wasn't a good option, as that would involve changing an existing 123 objects with those key-value pairs. I also need the the key-value pairs for other parts of my application.
My second-best thought was to create a standalone table to hold the keys that I need.
The solution I arrived at when talking with Scott is very specific to my data: the keys that I need all hold values of strings, and those strings are all either "yes" or "no", so I can check that to decide which key is rendered as a drop-down option.

**Comment by notdanross** [comment]
Solution:

- I set one useEffect to get the needed data
- I set another useEffect with a dependency array set by first useEffect
- Inside the second useEffect, an ```if``` statement double-checks the dependency array, then filters the desired data and sets a state variable.
